src/main/java/frc/robot/subsystems/amp/Amp.java

```diff
@@ -0,0 +1,157 @@
+package frc.robot.subsystems.amp;
+
+import static edu.wpi.first.units.Units.*;
+
+import edu.wpi.first.math.geometry.Pose3d;
+import edu.wpi.first.math.geometry.Rotation3d;
+import edu.wpi.first.math.geometry.Translation3d;
+import edu.wpi.first.wpilibj.smartdashboard.Mechanism2d;
+import edu.wpi.first.wpilibj.smartdashboard.MechanismLigament2d;
+import edu.wpi.first.wpilibj.smartdashboard.MechanismRoot2d;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import edu.wpi.first.wpilibj2.command.sysid.SysIdRoutine;
+import frc.robot.Constants;
+import java.util.Optional;
+import java.util.function.DoubleSupplier;
+import org.littletonrobotics.junction.Logger;
+
+// Subsystem base automatically runs periodic, too scared to put stuff in robot.java
+public class Amp extends SubsystemBase {
+
+  private final AmpIOInputsAutoLogged inputs = new AmpIOInputsAutoLogged();
+  private Mechanism2d ampBarPivot;
```

Comment:
Theres aren't needed unless you want to do a 2d sim. Def keep all the 3d sim stuff and modify it to fit the pivot bc that's like cool for 3d sim stuff.